ecis

```diff
@@ -127,18 +128,20 @@ function set_webchat_url {
 }
 
 function set_app_version_config {
-    sed -i "9s|.*|    APP_VERSION: '$1'|" $FRONTEND_CONFIG_FILE
+    tmpfile=$(mktemp)
+    sed "9s|.*|    APP_VERSION: '$1'|" $FRONTEND_CONFIG_FILE > "$tmpfile" && mv "$tmpfile" $FRONTEND_CONFIG_FILE
     catch_error $? "APP VERSION on Frontend $1"
-    sed -i "9s|.*|    APP_VERSION: '$1'|" $SUPPORT_CONFIG_FILE
+    sed "9s|.*|    APP_VERSION: '$1'|" $SUPPORT_CONFIG_FILE > "$tmpfile" && mv "$tmpfile" $SUPPORT_CONFIG_FILE
```

Comment:
Why remove -i?

Reply:
`sed` has a different implemention on unix based systems (linux) and BSD systems (mac os). While on linux it simply edits the file in-place if we don't pass a backup extension (optional), on BSD it expects it as a *mandatory* argument.
(`sed -i .bak ...` for instance)
In order to have a portable solution (and also one where we dont end up with several backup files cluterring the repository), we write sed output to a temp file, and then replace our original file with said temp file.

Linux sed:
```
-i[SUFFIX], --in-place[=SUFFIX]
    edit files in place (makes backup if extension supplied). The default operation mode is to break symbolic and hard links. This can be changed with --follow-symlinks and --copy. 
```

BSD sed:
```  
-i[extension]
    Edit files in place, saving backups with the specified extension. If a zero length extension is given, no backup will be saved. It is not recommended to give a zero length extension when in place editing files, as it risks corruption or partial content in situations where disk space is exhausted, etc. In -i mode, the hold space, line numbers, and ranges are reset between files.
```